src/main/webapp/app/service/trial.service.ts

```diff
@@ -96,23 +97,37 @@ export class TrialService {
         // prepare main types list
         this.connectionService.getMainType().subscribe((res: Array<string>) => {
             this.mainTypesOptions = this.mainTypesOptions.concat(res);
-            this.connectionService.getSubType().subscribe((response: Array<any>) => {
-                for (const item of response) {
-                    if (item.level !== 0) {
-                        const currentMaintype = item.mainType.name;
-                        const currentSubtype = item.name;
+            const mainTypeQueries = [];
+            for (const item of res) {
+                mainTypeQueries.push({
+                    'query': item,
+                    'version': this.oncotreeVersion,
+                    'type': 'mainType'
+                });
+            }
+            // prepare subtypes by maintype
+            const queries =  {
+                'queries': mainTypeQueries
+            };
+            this.connectionService.getSubType(queries).subscribe((response: Array<any>) => {
```

Comment:
When the API failed, we should give a message saying either queries invalid or the service is not available depends on the http code.